src/main/java/org/opensrp/service/PhysicalLocationService.java

```diff
@@ -526,5 +526,12 @@ public Long countAllLocations(Long serverVersion){
 	public Long countAllStructures(Long serverVersion){
 		return locationRepository.countAllStructures(serverVersion);
 	}
+
+	public LocationTree buildLocationHeirarchyWithAncestors(String locationId) {
+		LocationTree locationTree = new LocationTree();
+		Set<LocationDetail> locationDetails = locationRepository.findLocationWithAncestors(locationId);
+		locationTree.buildTreeFromList(getLocations(locationDetails, true));
+		return locationTree;
+	}
```

Comment:
Could we just return `Set<LocationDetail>` as this maybe easier for the FE team

Reply:
Sure